Source/CesiumRuntime/Private/CesiumGltfComponent.cpp

```diff
@@ -1168,10 +1168,13 @@ static void loadMesh(
 
   for (const CesiumGltf::MeshPrimitive& primitive : mesh.primitives) {
     CreatePrimitiveOptions primitiveOptions = {&options, &primitive};
-    loadPrimitive(
-        result->primitiveResults.emplace_back(),
-        transform,
-        primitiveOptions);
+    LoadPrimitiveResult loadPrimitiveResult;
+    loadPrimitive(loadPrimitiveResult, transform, primitiveOptions);
+
+    // if it doesn't have render data, then it can't be loaded
+    if (loadPrimitiveResult.RenderData) {
+      result->primitiveResults.emplace_back(loadPrimitiveResult);
+    }
```

Comment:
Just a nitpick. This causes a copy when inserting into primitiveResults when the render data exists right? Do you think it might be better if we keep the `loadPrimitive(result->primitiveResults.emplace_back(), ...)` but `pop_back` the last result out of the list if we find that the render data is missing afterwards? That way, in the usual case where render data exists, we don't create a copy. The potential realloc of the list might not matter if we call `vector.reserve(...)` in `loadMesh` (which I should have done when I made these changes).

Anyways, if you know the compiler optimizes the copy away somehow or feel this change wouldn't make a difference, feel free to resolve the comment and I'll merge!

Reply:
yup that works. I updated the change in the new commit